File: fixtures/scale/ruby/large/lib/check_hexpack.rb
require_relative 'hexpack_encode'
require_relative 'hexpack_decode'
require_relative 'corpus_codec'

def check_hexpack
  corpus_codec.each do |text|
    packed = hexpack_encode(text)
    return false if hexpack_decode(packed) != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_hexpack.py
from cipherlab.ciphers.hexpack import hexpack_encode, hexpack_decode
from cipherlab.specs.hexpack_spec import hexpack_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_hexpack():
    spec = hexpack_spec()
    for text in corpus_codec():
        packed = hexpack_encode(text)
        if hexpack_decode(packed) != text:
            return False
    return spec["category"] == "codec"
